aws/resource_aws_db_proxy_test.go

```diff
@@ -0,0 +1,274 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_db_proxy", &resource.Sweeper{
+		Name: "aws_db_proxy",
+		F:    testSweepRdsDbProxies,
+	})
+}
+
+func testSweepRdsDbProxies(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("Error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).rdsconn
+
+	err = conn.DescribeDBProxiesPages(&rds.DescribeDBProxiesInput{}, func(out *rds.DescribeDBProxiesOutput, lastPage bool) bool {
+		for _, dbpg := range out.DBProxies {
+			if dbpg == nil {
+				continue
+			}
+
+			input := &rds.DeleteDBProxyInput{
+				DBProxyName: dbpg.DBProxyName,
+			}
+			name := aws.StringValue(dbpg.DBProxyName)
+
+			log.Printf("[INFO] Deleting DB Proxy: %s", name)
+
+			_, err := conn.DeleteDBProxy(input)
+
+			if err != nil {
+				log.Printf("[ERROR] Failed to delete DB Proxy %s: %s", name, err)
+				continue
+			}
+		}
+
+		return !lastPage
+	})
+
+	if testSweepSkipSweepError(err) {
+		log.Printf("[WARN] Skipping RDS DB Proxy sweep for %s: %s", region, err)
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("Error retrieving DB Proxies: %s", err)
+	}
+
+	return nil
+}
+
+func TestAccAWSDBProxy_basic(t *testing.T) {
```

Comment:
This `_basic` test should include tests for all attributes defaults including:

```go
resource.TestCheckResourceAttr(resourceName, "auth.#", "1"),
tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "auth.*", map[string]string{
	"auth_scheme": "SECRETS",
	"description": "test",
	"iam_auth":    "DISABLED",
}),
resource.TestCheckResourceAttr(resourceName, "debug_logging", "false"),
resource.TestMatchResourceAttr(resourceName, "endpoint", regexp.MustCompile(`fill me in`)),
resource.TestCheckResourceAttr(resourceName, "idle_client_timeout", "1800"),
resource.TestCheckResourceAttrPair(resourceName, "role_arn", "aws_iam_role.test", "arn"),
resource.TestCheckResourceAttr(resourceName, "require_tls", "false"),
resource.TestCheckResourceAttr(resourceName, "vpc_subnet_ids.#", "2"),
tfawsresource.TestCheckTypeSetElemAttrPair(resourceName, "vpc_subnet_ids.*", "aws_subnet.test.0", "id"),
tfawsresource.TestCheckTypeSetElemAttrPair(resourceName, "vpc_subnet_ids.*", "aws_subnet.test.1", "id"),
```

Reply:
Looks like this was partially handled -- the placeholder `endpoint` regular expression still needs to be filled in.